src/solver/constraints-builder/cbuilder.cpp

```diff
@@ -87,15 +87,12 @@ bool CBuilder::update(bool applyCheckBox)
     Vector enabledACLines;
     pMesh.clear();
 
-    // update impedances from study file and computa impedance changes
+    // Update impedances from study file and compute impedance changes
 
     for (auto i = pLink.begin(); i != pLink.end(); i++)
     {
-        // try to open the file
-        if (!(*i)->dataLink->loadFromCSVFile((*i)->ptr->data.jit->sourceFilename,
-                                             Data::fhlMax,
-                                             (uint)8760,
-                                             Matrix<>::optImmediate))
+        // Try to open the file
+        if (not (*i)->ptr->loadDataFromCVSfile(Matrix<>::optImmediate))
```

Comment:
You're de-referencing two pointers in a single instruction. This is dangerous since one of them could be `nullptr` or invalid.

---

It's unclear what `ptr` refers to, consider renaming this variable.

Reply:
ok, **ptr** is a pointer to an object of type **AreaLink** (a study link actually).
For now, I do not rename that pointer, but I define a variable that makes this part of the code more clear.